src/vma/proto/route_entry.cpp

```diff
@@ -90,13 +90,12 @@ void route_entry::set_val(IN route_val* &val)
 
 void route_entry::register_to_net_device()
 {
-	net_dev_lst_t* dev_list = g_p_net_device_table_mgr->get_net_device_val_lst_from_index(m_val->get_if_index());
-	if (!dev_list || dev_list->empty()) {
+	local_ip_list_t lip_offloaded_list = g_p_net_device_table_mgr->get_ip_list(m_val->get_if_index());
+	if (lip_offloaded_list.empty()) {
 		rt_entry_logdbg("No matched net device for %s interface", m_val->get_if_name());
 		m_b_offloaded_net_dev = false;
-	}
-	else {
-		ip_address src_addr = dev_list->front()->get_local_addr();
+	} else {
+		ip_address src_addr = lip_offloaded_list.front().local_addr;
```

Comment:
Why do we take the front element here ? While ring_logic != ring_per_interface, it will attach the flow to the first ring